Core/automation/lib/python/core/metadata.py

```diff
@@ -0,0 +1,193 @@
+'''
+Wrapper class for using Item Metadata
+'''
+
+from collections import MutableMapping
+from org.eclipse.smarthome.core.items import Metadata, MetadataKey
+from core.log import logging, LOG_PREFIX
+log = logging.getLogger(LOG_PREFIX + ".metadata")
+from core import osgi
+MetadataRegistry = osgi.get_service("org.eclipse.smarthome.core.items.MetadataRegistry")
+
+__all__ = [ "namespace_exists", "item_metadata", "metadata_namespace" ]
+
+
+def namespace_exists(item_name, namespace):
+    '''Returns True if metadata namespace exists'''
+    return False if MetadataRegistry.get(MetadataKey(namespace, item_name)) is None else True
+
+def _resolveType(value):
+    '''Attempts to resolve the type of the metadata value.
+    It will return the value as the python type if possible,
+    otherwise will return value as string'''
+    if str(value).lower().strip() == "true":
+        return True
+    elif str(value).lower().strip() == "false":
+        return False
+    else:
+        # attempt to cast to int
+        try: return int(str(value))
+        except ValueError: pass
+        # attempt to cast to float
+        try: return float(str(value))
+        except ValueError: pass
```

Comment:
Sorry if this is nit picky, but I think these should go on separate lines for readability. There are some other places too.

Reply:
Consistency is important. This is my style, not the repo's. Thinking about it, I don't know where I picked up the habit of trying to keep the line count down

Reply:
I tend to do the same. In the past, I would try to condense everything as much as I could, but I'm becoming more used to the concept of readability.